### ISSUE TITLE
Get fastplotlib working in a docker container and binder or jupyterhub 

### ISSUE THREAD

**Issue by kushalkolar** [fastplotlib/fastplotlib#398]
It's important for us to fully figure out how to use `fastplotlib` in docker on machines with a GPU. @apasarkar has been battling this for some time and so far we suspect the issue is that the `nvidia-docker` container does not contain the necessary graphics libraries (libglx etc. and related libs) to run vulkan applications. We also think `libvulkan1` is required. We've tried installing some of these into the container with no luck so far (will post more details soon). We also suspect that the vulkan drivers might need to be installed on the host in order for the container to be able to use them.

Also tagging @billbrod since he has figured out Flatiron's binder and this is important for the FENS workshop.

related #376

**Comment by billbrod** [comment]
I wrote my notes for getting the nemos workshop working on Flatiron's binder [on the readme](https://github.com/flatironinstitute/nemos-workshop-feb-2024?tab=readme-ov-file#setting-up-binder). Binder [recommends against specifying the Dockerfile](https://mybinder.readthedocs.io/en/latest/tutorials/dockerfile.html) explicitly, and instead using some combination of the other [config files](https://mybinder.readthedocs.io/en/latest/using/config_files.html) they support.

I ended up getting it working with two conda environment files (`environment.yml` and `environment-cuda.yml`), plus a `postBuild` bash script. Binder sees the `environment.yml` file, uses conda to install it (which includes python, pip, cuda),  then runs the `postBuild` script, in which the first line installs is `CONDA_OVERRIDE_CUDA="11.8" conda env update -p $NB_PYTHON_PREFIX --file environment-cuda.yml` (this second file includes jax and all our other requirements; the rest of `postBuild` does nemos-specific things: downloading the data, building the notebooks, and moving things around). Being able to set `CONDA_OVERRIDE_CUDA` was the key to getting it working, putting everything in a single environment file didn't install jax with cuda support.

If you want to just have a binder that works on the Flatiron cluster, this should suffice for you, but if you're looking for a general docker solution, it's probably not enough.
